odoo/addons/base/models/ir_qweb.py

```diff
@@ -36,7 +405,77 @@
 ])) - _BLACKLIST
 
 
-class IrQWeb(models.AbstractModel, QWeb):
+# eval to compile generated string python code into binary code, used in `_compile`
+unsafe_eval = eval
+
+
+VOID_ELEMENTS = frozenset([
+    'area', 'base', 'br', 'col', 'embed', 'hr', 'img', 'input', 'keygen',
+    'link', 'menuitem', 'meta', 'param', 'source', 'track', 'wbr'])
+# Object available globally in functions compiled by the IrQWeb.
+AVAILABLE_OBJECTS = dict(_BUILTINS)
+# Terms allowed in addition to AVAILABLE_OBJECTS when compiling python expressions
+ALLOWED_KEYWORD = ['False', 'None', 'True', 'and', 'as', 'elif', 'else', 'for', 'if', 'in', 'is', 'not', 'or']
+# regexpr for string formatting and extract ( ruby-style )|( jinja-style  ) used in `_compile_format`
+FORMAT_REGEX = re.compile(r'(?:#\{(.+?)\})|(?:\{\{(.+?)\}\})')
+EMPTY_LINES_REGEX = re.compile(r'\n\s*\n')
+# Attribute name used outside the context of the QWeb.
+SPECIAL_DIRECTIVES = {'t-translation', 't-ignore', 't-title'}
+# Name of the variable to insert the content in t-call in the template.
+INNER_T_CALL = '0'
```

Comment:
Why 0? Maybe `slot` instead like a lot of JS frameworks. or a new directive `t-slot` ?